src/network/networkd-dhcp4.c

```diff
@@ -108,6 +107,12 @@ static int link_set_dhcp_routes(Link *link) {
                  * the addresses now, let's not configure the routes either. */
                 return 0;
 
+        r = set_ensure_allocated(&link->dhcp_routes, &route_full_hash_ops);
+        if (r < 0)
+                return log_oom();
+
+        set_clear_with_destructor(link->dhcp_routes, route_free);
```

Comment:
Maybe add a comment like `/* Clear old entries in case the set was already allocated */`?


If DEFINE_HASH_OPS_FULL() is used, specifying the destructor here would not be necessary.